Distraction Shield/background.js

```diff
@@ -0,0 +1,111 @@
+//Set that holds the urls to be intercepted
+var blockedSites = new BlockedSiteList();
+var interceptDateList = [];
+var localSettings = new UserSettings();
+
+/* --------------- ------ setter for local variables ------ ---------------*/
+
+setLocalSettings = function(newSettings) {
+    var oldState = localSettings.getState();
+    localSettings.copySettings(newSettings);
+    if (oldState != localSettings.getState()) {
+        replaceListener();
+    }
+};
+
+setLocalBlacklist = function(newList) {
+    blockedSites.setList(newList.getList());
+    replaceListener();
+};
+
+setLocalInterceptDateList = function(dateList) {
+    interceptDateList = dateList;
+};
+
+/* --------------- ------ Storage retrieval ------ ---------------*/
+// Methods here are only used upon initialization of the session.
+// Usage found in init.js
+
+retrieveSettings = function(callback, param) {
+    storage.getSettings(function(settingsObject) {
+        localSettings = settingsObject;
+        return callback(param);
+    });
+};
+
+retrieveBlockedSites = function(callback){
+    storage.getBlacklist(function(blacklist) {
+        blockedSites.setList(blacklist.getList());
+        return callback();
+    });
+};
+
+/* --------------- ------ Updating of variables ------ ---------------*/
+
+addUrlToBlockedSites = function(unformattedUrl, onSuccess) {
+    blockedSiteBuilder.createNewBlockedSite(unformattedUrl, function(newBS) {
+        if (blockedSites.addToList(newBS)) {
+            synchronizer.syncBlacklist(blockedSites);
+            onSuccess();
+        } 
+    });
+};
+
+/* --------------- ------ webRequest functions ------ ---------------*/
+
+replaceListener = function() {
+    removeWebRequestListener();
+    var urlList = blockedSites.getActiveUrls();
+    if (localSettings.getState() == "On" && urlList.length > 0) {
```

Comment:
We could have made it a boolean that was true or false, but showing "On" and "Off" seemed clearer